blog/zh/blog/2022/11/10/what-is-service-in-microservice-discovery.md

```diff
@@ -0,0 +1,122 @@
+---
+title: "微服务中的服务发现是什么"
+author: "罗泽轩"
+authorURL: "https://github.com/spacewander"
+authorImageURL: "https://github.com/spacewander.png"
+keywords: 
+- 微服务
+- 服务发现
+- 开源
+- APISIX
+- Nacos
+description: 本文通过服务发现的相关背景和 APISIX 对于服务发现的应用与实践，来介绍微服务中的服务发现内容。
+tags: [Ecosystem]
+---
+
+> 本文通过服务发现的相关背景和 APISIX 对于服务发现的应用与实践，来介绍微服务中的服务发现内容。
+
+<!--truncate-->
+
+## 服务发现是什么？为什么需要？
+
+在互联网刚开始出现的年代，人们要想访问某个在线服务，需要输入一长串的 IP 地址。IP 地址虽然不长，但是作为一串无意义的数字，要求记住特定服务的特定地址还是很考验人的记忆力。所以后来人们就发明了域名系统。每个在线服务会到域名商注册一个域名，然后通过 DNS 建立域名和具体 IP 的联系。这样一来，人们只需要输入一个好记的域名，就能访问到具体 IP 上的在线服务。这就是最早的服务发现。
+
+当一个公司内部的服务数到了一定的规模（比如在做了微服务拆分之后），也会遇到 IP 实在不好记的问题。这时候就需要有一套服务发现系统。公司里面各个服务在该系统上注册，然后想要访问这些服务的其他服务会从该系统上查询对应的 IP 地址，这样就不需要让某个服务“记住”复杂多变的 IP 地址了。
+
+如下图，IP 地址的变更，会让访问者无所适从。
+
+![IP 变更](https://static.apiseven.com/2022/10/16/634bf34909674.png)
```

Comment:
You need to update the image, the current one has a grey line on the left side. Just crop that out.
<img width="912" alt="image" src="https://user-images.githubusercontent.com/36651058/201622311-fb4483e8-0078-49ba-8110-c21c54f0c342.png">

Reply:
done